sdk/program/src/vote/state/mod.rs

```diff
@@ -221,16 +221,21 @@ impl VoteStateUpdate {
 
 #[cfg_attr(
     feature = "frozen-abi",
-    frozen_abi(digest = "5PFw9pyF1UG1DXVsw7gpjHegNyRycAAxWf2GA9wUXPs5"),
+    frozen_abi(digest = "DqMoZ6hSsG1wguKx49kjZmU8sEu7jjzDWqg14qxoeinM"),
     derive(AbiExample)
 )]
 #[derive(Serialize, Default, Deserialize, Debug, PartialEq, Eq, Clone)]
 pub struct TowerSync {
-    /// The proposed tower
+    /// the proposed tower
     pub lockouts: VecDeque<Lockout>,
-    /// The proposed root
+    /// the proposed root
     pub root: Option<Slot>,
-    /// signature of the bank's state at the last slot
+    /// signature of the VoteState's at the latest slot in `lockouts`
+    pub yolo_hash: Hash,
+    /// the latest fully replayed slot, equal to the last
+    /// slot in `lockouts` until asynchronous execution.
+    pub replay_tip: Slot,
```

Comment:
I'm now thinking: for easy of rollout and debug, maybe we should add ephemeral_slot instead, ephemeral_slot is where the ephemeral_hash is calculated. So that we don't break anyone's tools which associate bankhash with the slot in the TowerSync vote transaction. We do have to change consensus code, but that's internal and easier to upgrade.

Reply:
we already have this value as the `last_voted_slot()`, and it shouldn't break anyone's tooling since `TowerSync` is a new instruction. Also `replay_tip` will equal `last_voted_slot()` until yolo-ape is turned on.

in terms of internal tooling it's better to break and fix it now rather than some future date when this feature is turned on.

Reply:
I think I'm mostly worried about that hash originally was associated with slot, now suddenly it's the hash of replay_tip. It's easy to fix internal tools, but not so much for outsiders who inspect vote instructions on chain. Instead of making outside observers realize that if we have replay_tip, they should associate hash with replay_tip, I think it is safer to just introduce new fields in pairs: yolo_hash/yolo_slot or ephemeral_hash/ephemeral_slot. The ephemeral hash/slot is like optimistic signal which will arrive faster to help you confirm a fork, but if you insist on only using hash/slot, you can do that.

Reply:
it doesn't make sense to add bits for duplicate info already present in the ix. if we're in the business of adding bits I'll gladly take a 64 for the ref slot instead 🙂.

again I want to iterate that this is a *new* instruction. any previous tooling attempting to deserialize this as a VoteStateUpdate will fail. if we want to add some extra method in the public API we can go ahead or im fine renaming hash to something like non-yolo-bank-hash or replay_tip_hash in the API if that makes it clearer.

Reply:
Nah I'm not asking to add additional bits for duplicate info in the ix. Just there are two hashes and I hope it's clearer from the names which hash corresponds to which slot.

Does name string take space in the serialized ix? We can make them shorter.

Reply:
I'm confused. what exactly is it you are proposing?

the initial message suggested adding an ephemeral slot and the third message suggested a yolo_slot and your most recent message asks about shortening hash names? 

perhaps it would be clearer if you typed out what you envision the TowerSync struct to look like?

Reply:
Sorry for the confusion:
I think ideally we could pair the two hash with slots:
```
pub optimistic_hash: Hash,
pub optimistic_slot: Slot,
pub replay_tip_hash: Hash,
pub replay_tip_slot: Slot,

```
or 

```
pub optimistic_hash: Hash,
pub optimistic_slot: Slot,
pub hash: Hash,
pub slot: Slot,

```
As long as it's obvious which slot a hash corresponds to, I don't care that much what name we use.

Reply:
ah got it just a rename. ya that makes sense will do once we decide on the final naming.

Reply:
Also, should replay_tip_slot be an option? Since root can also be an option.